src/snowflake/snowpark/types.py

```diff
@@ -552,29 +569,46 @@ def __init__(
         column_identifier: Union[ColumnIdentifier, str],
         datatype: DataType,
         nullable: bool = True,
+        is_column: bool = True,
```

Comment:
this parameter will be part of the public api, do we expect customer to set this?

Reply:
I do not expect anyone to set this. Can make it a private parameter.